src/call_builder.ts

```diff
@@ -139,18 +174,19 @@ export class CallBuilder {
    * @param {bool} [link.templated] Whether the link is templated
    * @returns {function} A function that requests the link
    */
-  _requestFnForLink(link) {
-    return (opts) => {
+  private _requestFnForLink(link: Horizon.ResponseLink): Function {
+    return async (opts: any) => {
       let uri;
 
       if (link.templated) {
         const template = URITemplate(link.href);
-        uri = URI(template.expand(opts || {}));
+        uri = URI(template.expand(opts || {}) as any);  // TODO: fix upstream types.
       } else {
         uri = URI(link.href);
       }
 
-      return this._sendNormalRequest(uri).then((r) => this._parseResponse(r));
+      const r = await this._sendNormalRequest(uri);
+      return this._parseResponse(r);
```

Comment:
I'm guessing this is because you prefer the async syntax to Promises? I don't share that preference, I think whatever makes the code more clear is better. In this particular case, I think the before and after are the same clarity, so I'm okay with your change.